receiver/kubeletstatsreceiver/scraper.go

```diff
@@ -67,8 +82,26 @@ func newKubletScraper(
 			metricsConfig.Metrics.K8sPodMemoryRequestUtilization.Enabled ||
 			metricsConfig.Metrics.K8sContainerMemoryLimitUtilization.Enabled ||
 			metricsConfig.Metrics.K8sContainerMemoryRequestUtilization.Enabled,
+		stopCh:   make(chan struct{}),
+		nodes:    make(map[string]kubelet.NodeLimits),
+		nodeName: nodeName,
 	}
-	return scraperhelper.NewScraper(metadata.Type.String(), ks.scrape)
+
+	if metricsConfig.Metrics.K8sContainerCPUNodeLimitUtilization.Enabled {
+		if nodeName == "" {
+			ks.logger.Warn("node name is not set. k8s.container.cpu.node_limit_utilization will not be calculated. Check the readme on how to set the required setting")
```

Comment:
We should enforce this validation in `config.go`'s Validate function (we might need to add one).

Reply:
Good point! Added at f7ac447